openverse_catalog/dags/providers/provider_csv_load_scripts/inaturalist/transformed_table.template.sql

```diff
@@ -0,0 +1,78 @@
+/*
+-------------------------------------------------------------------------------
+Load Intermediate Table
+-------------------------------------------------------------------------------
+
+    ** Please note: This SQL will not run as is! You must replace offset_num and
+    batch_limit with integers representing the records you want to retrieve.
+
+Joining two very large normalized tables is difficult, in any data manipulation system.
+PHOTOS has on the order of 120 million records, and OBSERVATIONS has on the order of
+70 million records. We have to join them to get at the taxa (species) information for
+any given photo. Taxa are the only descriptive text we have for inaturalist photos.
+
+Using image columns version 001 from common.storage.tsv_columns.
+*/
+
+INSERT INTO {intermediate_table}
+(
+    SELECT
+        /*
+        The same photo_id can have multiple records, for example if there are multiple
+        observations for separate species. But it only actually happens in about 0.1% of
+        the photos. For now, we're skipping them.
+        TO DO #685: Figure out aggregating tags and titles or formatting alternate
+        foreign ids for photos with multiple taxa and process them separately.
+        (If we go the alternate foreign id way, we'd want to drop photos loaded in the
+        first inaturalist load.)
+        */
+        INATURALIST.PHOTOS.PHOTO_ID as FOREIGN_IDENTIFIER,
+        'https://www.inaturalist.org/photos/' || INATURALIST.PHOTOS.PHOTO_ID
+            as LANDING_URL,
+        'https://inaturalist-open-data.s3.amazonaws.com/photos/'
+        || INATURALIST.PHOTOS.PHOTO_ID || '/medium.' || INATURALIST.PHOTOS.EXTENSION
+            as DIRECT_URL,
+        -- TO DO #810: Add the thumbnail url here.
+        null::varchar(10) as THUMBNAIL,
+        -- only jpg, jpeg, png & gif in 6/2022 data, all in extensions.py for images
+        lower(INATURALIST.PHOTOS.EXTENSION) as FILETYPE,
+        null::int as FILESIZE,
+        INATURALIST.LICENSE_CODES.OPENVERSE_CODE as LICENSE,
+        INATURALIST.LICENSE_CODES.LICENSE_VERSION,
+        COALESCE(INATURALIST.OBSERVERS.LOGIN, INATURALIST.PHOTOS.OBSERVER_ID::text)
+            as CREATOR,
```

Comment:
The name of the observer will be preferred over their username.
```suggestion
        COALESCE(INATURALIST.OBSERVERS.NAME, INATURALIST.OBSERVERS.LOGIN)
            as CREATOR,
```

Reply:
Ooh! Great catch! I think I must have had a left join to observers in a prior version and maybe I was worried about some names being truncated or something? I don't know. Hopefully my fix doesn't over-complicate things.